### ISSUE TITLE
Getting multiple errors in GraphQL from Formie since updating Craft to 3.7 and Formie to 1.5.5

### ISSUE THREAD

**Issue by toddpadwick** [verbb/formie#714]
**Description**
Since updating Craft to 3.7.23 and Formie to 1.5.5, when testing existing GraphQL queries in GraphiQL, I receive the following errors. (I have not changed any code and all was working fine before the update)

So far it seems to still work with Apollo, but I am still testing and yet to determine how much damage these errors are causing.

```
{
  "errors": [
    "Interface field EntryInterface.author expects type UserInterface but testimonials_testimonial_Entry.author is type String.",
    "Interface field RowInterface.id expected but feedback_KPIs_FormieRepeaterFieldRow does not provide it.",
    "Interface field RowInterface.uid expected but feedback_KPIs_FormieRepeaterFieldRow does not provide it.",
    "Interface field RowInterface.fields expected but feedback_KPIs_FormieRepeaterFieldRow does not provide it.",
    "Interface field RowInterface.id expected but testForm_repeater_FormieRepeaterFieldRow does not provide it.",
    "Interface field RowInterface.uid expected but testForm_repeater_FormieRepeaterFieldRow does not provide it.",
    "Interface field RowInterface.fields expected but testForm_repeater_FormieRepeaterFieldRow does not provide it.",
    "Interface field RowInterface.id expected but testForm_repeaterField_FormieRepeaterFieldRow does not provide it.",
    "Interface field RowInterface.uid expected but testForm_repeaterField_FormieRepeaterFieldRow does not provide it.",
    "Interface field RowInterface.fields expected but testForm_repeaterField_FormieRepeaterFieldRow does not provide it."
  ]
}
```

**Comment by engram-design** [comment]
Are you able to shoot through the full query you're using? I don't get those for my testing queries, but maybe you're including the `id` and `uid` (which should be fine).

**Comment by toddpadwick** [comment]
This error is appearing in the GraphiQL even when no formie query is submitted. It doesn't seem to correlate with the query in the editor at all.

This was the query that was in the editor at the time, but the formie error seems to return no matter what query is run:
```
query getEntry($slug: [String], $uri: [String], $section: [String], $id: [QueryArgument]) {
  entry(slug: $slug, section: $section, id: $id, uri: $uri) {
    title
    slug
    id
    uri
    typeHandle
    url
    status
    postDate
    dateCreated
    dateUpdated
    ... on homepage_homepage_Entry {
      heroBuilder {
        ...heroBuilder
      }
      landingPageBuilder {
        ...landingPageBuilder
      }
    }
    ... on blog_blog_Entry {
      headline
      description
    }
    ... on about_about_Entry {
      description
      heroBuilder {
        ...heroBuilder
      }
      landingPageBuilder {
        ...landingPageBuilder
      }
    }
    ... on services_service_Entry {
      description
      headline
      status
      heroBuilder {
        ...heroBuilder
      }
      landingPageBuilder {
        ...landingPageBuilder
      }
    }
    ... on caseStudies_caseStudy_Entry {
      headline
      introduction
      externalLink
      passwordProtect
      services(status: ["live", "disabled"]) {
        title
        url
        uri
        slug
        status
      }
      client {
        title
        slug
        ... on clients_client_Entry {
          location
          externalLink
        }
      }
      teaserImage {
        title
        width
        height
        url @transform(width: 300)
        srcset(sizes: ["300", "600", "960", "1200"])
      }
      screenRecording {
        ... on screenRecording_BlockType {
          id
          image {
            width
            height
            srcset(sizes: ["300", "600", "960", "1200"])
            title
            url @transform(width: 600, quality: 50)
          }
          video {
            height
            width
            title
            url
            mimeType
          }
        }
      }
      landingPageBuilder {
        ...landingPageBuilder
      }
    }
    next(section: "caseStudies", passwordProtect: false) {
      slug
      id
      url
      uri
      ... on caseStudies_caseStudy_Entry {
        passwordProtect
        headline
        client {
          title
          slug
        }
      }
    }
    ... on articles_article_Entry {
      title
      uri
      id
      slug
      description
      nextArticle: next(section: "articles") {
        slug
        id
        url
        title
        uri
      }
      previousArticle: prev(section: "articles") {
        slug
        id
        title
        url
        uri
      }
      heroImage {
        width
        height
        title
        srcset(sizes: ["300", "750", "1200", "1920"])
        url @transform(width: 750)
        id
      }
      articleBuilder {
        ...articleBuilderFrag
      }
      category {
        title
        uri
        id
        slug
        url
      }
      author {
        name
        firstName
        lastName
        id
        ... on User {
          shortDescription
          photo {
            srcset(sizes: ["48", "72", "96"])
            width
            height
            url @transform(width: 96)
            title
          }
        }
      }
    }
    ... on careers_permanent_Entry {
      headline
      description
      body
      skills {
        title
      }
    }
  }
}

fragment heroBuilder on heroBuilder_MatrixField {
  ... on heroBuilder_textOnly_BlockType {
    typeHandle
    preTitle
    headline
    introduction
    scrollArrowLabel
    size
    qualitySeal
    cta {
      ... on cta_BlockType {
        weight
        label
        entry {
          slug
          title
          sectionHandle
          url
          uri
        }
      }
    }
    backgroundText {
      ... on backgroundText_BlockType {
        text
      }
    }
  }
  ... on heroBuilder_images_BlockType {
    typeHandle
    scrollArrowLabel
    images {
      width
      height
      srcset(sizes: ["300", "600", "960", "1200"])
      title
      url @transform(width: 600)
    }
    backgroundText {
      ... on backgroundText_BlockType {
        text
      }
    }
  }
  ... on heroBuilder_textAndVisual_BlockType {
    typeHandle
    preTitle
    headline
    scrollArrowLabel
    image {
      width
      height
      srcset(sizes: ["300", "600", "960", "1200", "1920"])
      title
      url @transform(width: 600)
    }
    addShadow
    backgroundText {
      ... on backgroundText_BlockType {
        text
      }
    }
    cta {
      ... on cta_BlockType {
        weight
        label
        entry {
          slug
          title
          sectionHandle
          url
          uri
        }
      }
    }
  }
}

fragment landingPageBuilder on landingPageBuilder_MatrixField {
  ... on landingPageBuilder_introduction_BlockType {
    typeHandle
    preTitle
    headline
    background
    text
    sidebar
  }
  ... on landingPageBuilder_text_BlockType {
    typeHandle
    background
    text
  }
  ... on landingPageBuilder_stats_BlockType {
    typeHandle
    background
    stats {
      ... on stats_BlockType {
        label
        data
      }
    }
  }
  ... on landingPageBuilder_caseStudies_BlockType {
    background
    typeHandle
    entries {
      slug
      id
      url
      uri
      typeHandle
      sectionHandle
      ... on testimonials_testimonial_Entry {
        client {
          title
          slug
          ... on clients_client_Entry {
            externalLink
          }
        }
        author
        jobTitle
        quote
      }
      ... on caseStudies_caseStudy_Entry {
        headline
        passwordProtect
        services(status: ["live", "disabled"]) {
          title
          url
          uri
          slug
          status
        }
        client {
          title
        }
        teaserImage {
          title
          width
          height
          url @transform(width: 300)
          srcset(sizes: ["300", "600", "960", "1200"])
        }
        screenRecording {
          ... on screenRecording_BlockType {
            id
            image {
              width
              height
              srcset(sizes: ["300", "600", "960", "1200"])
              title
              url @transform(width: 600, quality: 50)
            }
            video {
              height
              width
              title
              url
              mimeType
            }
          }
        }
      }
    }
  }
  ... on landingPageBuilder_steps_BlockType {
    typeHandle
    background
    stepBlocks: steps {
      ... on steps_BlockType {
        text
      }
    }
  }
  ... on landingPageBuilder_gallery_BlockType {
    typeHandle
    background
    display
    images {
      width
      height
      title
      srcset(sizes: ["300", "750", "1200", "1920"])
      url @transform(width: 750)
      id
    }
  }
  ... on landingPageBuilder_columns_BlockType {
    typeHandle
    background
    columnBlocks: columns {
      ... on columns_BlockType {
        text
        image {
          width
          height
          srcset(sizes: ["300", "750", "1200"])
          title
          url @transform(width: 600)
        }
        cta {
          ... on cta_button_BlockType {
            label
            weight
            entry {
              url
              uri
              slug
              id
            }
          }
        }
      }
    }
  }
  ... on landingPageBuilder_partners_BlockType {
    typeHandle
    headline
    columns
    partners {
      ... on partners_partner_Entry {
        logo {
          width
          height
          srcset(sizes: ["100", "200", "300"])
          title
          url @transform(width: 100)
        }
      }
    }
  }
  ... on landingPageBuilder_articles_BlockType {
    typeHandle
    background
    category {
      slug
      id
      groupHandle
      uri
      title
    }
  }
  ... on landingPageBuilder_logoParade_BlockType {
    typeHandle
    background
    logos {
      width
      height
      title
      srcset(sizes: ["100", "200", "300"])
      url @transform(width: 100)
      id
    }
  }
  ... on landingPageBuilder_testimonial_BlockType {
    typeHandle
    background
    testimonial {
      ... on testimonials_testimonial_Entry {
        client {
          title
          slug
          ... on clients_client_Entry {
            externalLink
          }
        }
        author
        jobTitle
        quote
      }
    }
  }
  ... on landingPageBuilder_accreditations_BlockType {
    typeHandle
    background
    columns
    accreditations {
      ... on accreditations_accreditation_Entry {
        title
        id
        slug
        caption
        externalLink
        logo {
          width
          height
          title
          srcset(sizes: ["100", "200", "300"])
          url @transform(width: 100)
          id
        }
      }
    }
  }
  ... on landingPageBuilder_annotatedVisual_BlockType {
    typeHandle
    background
    numbered
    text
    caption
    shadow
    screenRecordingImage {
      width
      height
      srcset(sizes: ["300", "600", "960", "1200"])
      title
      url @transform(width: 600, quality: 50)
      id
    }
    screenRecordingVideo {
      width
      height
      mimeType
      url
      id
      title
    }
    image {
      width
      height
      url @transform(width: 600, quality: 50)
      srcset(sizes: ["300", "600", "960", "1200"])
      title
      id
    }
    loopingVideo {
      width
      height
      mimeType
      url
      id
      title
    }
  }
  ... on landingPageBuilder_screenRecording_BlockType {
    typeHandle
    caption
    screenRecordingImage {
      width
      height
      srcset(sizes: ["300", "600", "960", "1200"])
      title
      url @transform(width: 600, quality: 50)
      id
    }
    screenRecordingVideo {
      width
      height
      mimeType
      url
      id
      title
    }
  }
  ... on landingPageBuilder_sectionHeadline_BlockType {
    typeHandle
    headline
    introduction
  }
  ... on landingPageBuilder_keyPoints_BlockType {
    typeHandle
    ordered
    points {
      ... on points_BlockType {
        label
        description
      }
    }
  }
  ... on landingPageBuilder_careers_BlockType {
    typeHandle
    careers {
      title
      typeHandle
      uri
      url
      id
      slug
      ... on careers_permanent_Entry {
        headline
        description
        skills {
          title
        }
      }
    }
  }
}

fragment articleBuilderFrag on articleBuilder_MatrixField {
  ... on articleBuilder_sectionHeadline_BlockType {
    typeHandle
    headline
  }
  ... on articleBuilder_richText_BlockType {
    typeHandle
    text
  }
  ... on articleBuilder_callToAction_BlockType {
    typeHandle
    caption
    headline
    button {
      ... on button_BlockType {
        label
        weight
        entry {
          title
          slug
          uri
          id
          url
        }
      }
    }
  }
  ... on articleBuilder_image_BlockType {
    typeHandle
    image {
      width
      id
      height
      title
      url @transform(width: 1920)
      extension
      srcset(sizes: ["750", "1024", "1920"])
      ... on blog_Asset {
        caption
      }
    }
  }
  ... on articleBuilder_video_BlockType {
    typeHandle
    video {
      width
      height
      size
      extension
      url
      id
      mimeType
    }
  }
  ... on articleBuilder_blockQuote_BlockType {
    typeHandle
    quote
    author
    role
    organisation
  }
  ... on articleBuilder_link_BlockType {
    typeHandle
    linkUrl
    linkTitle
  }
  ... on articleBuilder_screenRecording_BlockType {
    typeHandle
    caption
    screenRecordingImage {
      width
      height
      srcset(sizes: ["300", "600", "960", "1200"])
      title
      url @transform(width: 600, quality: 50)
      id
    }
    screenRecordingVideo {
      width
      height
      mimeType
      url
      id
      title
    }
  }
}
```

**Comment by engram-design** [comment]
Oh, I see, I thought it might've been the query. I might chat with @andris-sevcenko about it, in case I'm doing something wrong.

**Comment by mfeichtmeier** [comment]
Hi, 
we're having the same issue with both formie 1.5.5 and 1.5.9, with Craft 3.7.28.

```
{
  "errors": [
    "Interface field RowInterface.id expected but appointment_registration_appointmentParticipant_FormieRepeaterFieldRow does not provide it.",
    "Interface field RowInterface.uid expected but appointment_registration_appointmentParticipant_FormieRepeaterFieldRow does not provide it.",
    "Interface field RowInterface.fields expected but appointment_registration_appointmentParticipant_FormieRepeaterFieldRow does not provide it."
  ]
}
```

schema.graphql

<img width="838" alt="image" src="https://user-images.githubusercontent.com/59869087/155171501-9b29396b-c594-4a4e-8ce9-f1f174984f7f.png">

![formieRepeaterFieldRow](https://user-images.githubusercontent.com/59869087/155172282-946cb3ae-0b61-4a94-a786-e16c750cc5b9.png)

**Comment by benediktblank** [comment]
I was having the exact same problem: As soon as I use a repeater element in a form, GraphQL starts behaving weird and errors are showing up.

Even when I have very simple form, it starts as soon as you use a repeater element:
![image](https://user-images.githubusercontent.com/69143768/156188009-eba8d0d2-1f00-4df9-8e84-f5888a0a2c84.png)

In the CraftCMS graphiql console these errors start appearing:
```
{
  "errors": [
    "Interface field RowInterface.id expected but test1_testrepeater_FormieRepeaterFieldRow does not provide it.",
    "Interface field RowInterface.uid expected but test1_testrepeater_FormieRepeaterFieldRow does not provide it.",
    "Interface field RowInterface.fields expected but test1_testrepeater_FormieRepeaterFieldRow does not provide it."
  ]
}
```
![image](https://user-images.githubusercontent.com/69143768/156188278-7da725ef-336e-495f-8560-c1b2adbbec55.png)

This also breaks the auto complete feature in the CraftCMS graphiql console.

I've had a look in the code and I found a possible solution in this file:
vendor/verbb/formie/src/fields/formfields/Repeater.php

In the getContentGqlType function, I've made some adjustments:

I've commented out this:
```
        $rowType = GqlEntityRegistry::createEntity($rowTypeName, new RowType([
            'name' => $rowTypeName,
            'fields' => function() use ($repeaterFields) {
                return $repeaterFields;
            },
        ]));
```
And replaced it with:
```
        $fieldFields = RowInterface::getFieldDefinitions();
        $new = array_merge($repeaterFields, $fieldFields);

        $rowType = GqlEntityRegistry::createEntity($rowTypeName, new RowType([
            'name' => $rowTypeName,
            'fields' => function() use ($new) {
                return $new;
            },
        ]));
```
RowInterface is: verbb\formie\gql\interfaces\RowInterface

That way the missing interface fields id, uid and fields are being added.
That fixed the error and restores the auto complete ability as well.

@engram-design 
Can you please have a look at this and maybe include it in the next release?
Or maybe you find a better solution based on what I found out.

**Comment by engram-design** [comment]
Thanks @benediktblank I was actually getting @andris-sevcenko to give me a hand with this, as it seems to be happening for some native Craft fields too. I'll pass this on to see if this is the correct way to go about it, but can also confirm this removes those errors.

Fixed for the next release. To get this fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.5.10",
  "...": "..."
}
```

Then run `composer update`.

**Comment by toddpadwick** [comment]
I have also noticed that formie fields relating to categories, tags or entries don't work and break GraphQL queries (the error messages are limited and don't provide any context). To get it to work, I have simply commented out the following – which for the time being isn't an issue because I am not currently using these fields:
```
  ... on Field_Categories {
        displayType
  
        categories {
            id
            title
        }
    }
  
  ... on Field_Entries {
      displayType
      entries {
          id
          title
      }
  }
  
  ... on Field_Tags {
      displayType
  
      tags {
          id
          title
      }
  }
```

Could this be related?

**Comment by engram-design** [comment]
@toddpadwick What's the error you're getting? Nothing seems to report back on my end, and does query successfully .

**Comment by engram-design** [comment]
Fixed in [1.5.11](https://github.com/verbb/formie/releases/tag/1.5.11)
